contracts/base.sol

```diff
@@ -7,12 +7,35 @@ contract DSBaseNoFallback {
         log1(msg.sig, reason);
         throw;
     }
+    // this may be default in future: https://github.com/ethereum/solidity/pull/665
     modifier noEther() {
-        if( msg.value == 0 ) {
-            _
-        } else {
-            throw;
-        }
+        assert( msg.value == 0 );
+        _
+    }
+    // N.B. In older solidity versions, this cannot be used on functions
+    // with a `return` statement. This restriction is lifted now
+    // (https://github.com/ethereum/solidity/commit/9c8310)
+    bool _mutex;
+    modifier mutex() {
```

Comment:
I'm inclined to somehow make that more clear, but I think the best we can do overall is declare all solidity source prior to the required version pragma to be unsafe